crates/wasi-common/src/hostcalls_impl/fs.rs

```diff
@@ -485,7 +485,7 @@ pub(crate) unsafe fn fd_write(
 }
 
 pub(crate) unsafe fn fd_advise(
-    wasi_ctx: &WasiCtx,
+    wasi_ctx: &mut WasiCtx,
```

Comment:
This is something that I was uncomfortable about with `OsHandleRef` - it very easily lets us lie about mutability of file handles. Because OS handles are wrappers around an integer, I'm not sure _how_ dangerous this ends up being but by not `OsHandleRef`'ing around immutability of files, this `wasi_ctx` must be `&mut`.

Reply:
FWIW, when @pchickey and myself finish the first fully working draft of `wig`'s upgrade [`wiggle`](https://github.com/kubkon/wiggle), all hostcalls will require `&mut` to the context entity. Borrowing rules (accesses to guest's memory and such) will on the other hand be enforced at runtime.